src/game/common/bitflags.h

```diff
@@ -242,77 +439,89 @@ template<int bits> void BitFlags<bits>::Parse(INI *ini, Utf8String *string)
+            }
+
+            string->Concat(token);
+        }
 
-                *string += token;
+        if (!strcasecmp(token, "NONE")) {
+            if (set || adjust) {
+                captainslog_error("you may not mix normal and +- ops in bitstring lists");
+                throw 0xDEAD0006;
             }
 
-            // If we have a plus or minus as first char, then that decides if bit is set or cleared.
-            // Otherwise just set.
-            if (*token == '+') {
-                captainslog_relassert(!set,
-                    0xDEAD0006,
-                    "File: '%s', Line: %d Trying to adjust when we already set.",
-                    ini->Get_Filename().Str(),
-                    ini->Get_Line_Number());
-
-                m_bits.set(INI::Scan_IndexList(token + 1, s_bitNamesList));
-                adjust = true;
-            } else if (*token == '-') {
-                captainslog_relassert(!set,
-                    0xDEAD0006,
-                    "File: '%s', Line: %d Trying to adjust when we already set.",
-                    ini->Get_Filename().Str(),
-                    ini->Get_Line_Number());
-
-                m_bits.reset(INI::Scan_IndexList(token + 1, s_bitNamesList));
-                adjust = true;
-            } else {
-                captainslog_relassert(!adjust,
-                    0xDEAD0006,
-                    "File: '%s', Line: %d Trying to set when we already adjusted.",
-                    ini->Get_Filename().Str(),
-                    ini->Get_Line_Number());
-
-                // If we haven't started setting yet, 0 it out.
-                if (!set) {
-                    m_bits.reset();
-                }
+            Clear();
+            return;
+        }
 
-                m_bits.set(INI::Scan_IndexList(token, s_bitNamesList));
-                set = true;
+        // If we have a plus or minus as first char, then that decides if bit is set or cleared.
+        if (*token == '+') {
+            if (set) {
+                captainslog_error("you may not mix normal and +- ops in bitstring lists");
+                throw 0xDEAD0006;
             }
 
-            token = ini->Get_Next_Token_Or_Null();
+            Set(INI::Scan_IndexList(token + 1, s_bitNamesList), 1);
+            adjust = true;
+        } else if (*token == '-') {
+            if (set) {
+                captainslog_error("you may not mix normal and +- ops in bitstring lists");
+                throw 0xDEAD0006;
+            }
 
-            if (token == nullptr) {
-                return;
+            Set(INI::Scan_IndexList(token + 1, s_bitNamesList), 0);
+            adjust = true;
+        } else {
+            if (adjust) {
+                captainslog_error("you may not mix normal and +- ops in bitstring lists");
+                throw 0xDEAD0006;
             }
+
+            if (!set) {
+                Clear();
+            }
+
+            Set(INI::Scan_IndexList(token + 1, s_bitNamesList), 1);
```

Comment:
token should not be incremented by 1. There is no operator so it does not need to be skipped.

Reply:
fixed.